### PR TITLE
Remove dependency on usmap by updating bubble plotting and removing county_geo/state_geo

### PR DESCRIPTION
**Pull request by sarah-colq**
Update the bubble plots to use the shapefiles for state and county instead of usmap.   This also adds PR.  

Removes usmap as a dependency by removing state_geo and county_geo functions and documentation.  

Update dependency documentation.  

This will fix issues #420 and #24 .

### CODE REVIEW

File: R-packages/covidcast/R/plot.R [hunk 1]
```diff
@@ -582,69 +582,130 @@ plot_bubble = function(x, time_value = NULL, include = c(), range = NULL,
...
+  
+  # Explicitly drop zeros (and from levels) unless were asked not to
+  if (!isFALSE(params$remove_zero)) {
+    map_df$bubble_val[map_df$bubble_val == 0] = NA
+    levels(map_df$bubble_val)[levels(map_df$bubble_val) == 0] = NA
+  }
+  
   # Warn if there's any missing locations
-  if (sum(map_mis == 1) > 0) {
+  if (sum(map_df$back_color == missing_col) > 0) {
```
Comment (capnrefsmmat) [inline R-packages/covidcast/R/plot.R:649]:
```suggestion
  if (any(map_df$back_color == missing_col)) {
```

File: R-packages/covidcast/R/data.R [hunk 2]
```diff
@@ -73,42 +73,4 @@
...
-#'   \item{x}{Longitude of state centroid.}
-#'   \item{y}{Latitude of state centroid.}
-#'   \item{fips}{Five-digit county FIPS code.}
-#'   \item{abbr}{Two-letter state abbreviation.}
-#'   \item{full}{State name.}
-#' }
-#'
-#' @source `usmap`
-"state_geo"
+"state_census"
```
Comment (capnrefsmmat) [inline R-packages/covidcast/R/data.R:76]: newline at end of file

File: R-packages/covidcast/R/plot.R [hunk 3]
```diff
@@ -582,69 +582,130 @@ plot_bubble = function(x, time_value = NULL, include = c(), range = NULL,
...
+  geom_args$fill = hawaii_df$back_color
+  geom_args$data = hawaii_df
+  hawaii_layer = do.call(ggplot2::geom_sf, geom_args)
+  geom_args$fill = alaska_df$back_color
+  geom_args$data = alaska_df
+  alaska_layer = do.call(ggplot2::geom_sf, geom_args)
+  
+  # Change geometry to centroids
+  # Use centroid coordinates for plotting bubbles
+  suppressWarnings({
```
Comment (capnrefsmmat) [inline R-packages/covidcast/R/plot.R:681]: Why is `suppressWarnings` needed here? Does `st_centroid` raise warnings? If it does, and we need to ignore them, we should have a comment saying what the warning is and why it's safe to ignore

File: R-packages/covidcast/R/plot.R [hunk 4]
```diff
@@ -582,69 +582,130 @@ plot_bubble = function(x, time_value = NULL, include = c(), range = NULL,
...
+    pr_df$geometry <- sf::st_centroid(pr_df$geometry)
+  })
 
-  # Explicitly drop zeros (and from levels) unless we're asked not to
-  if (!isFALSE(params$remove_zero)) {
-    cur_val[cur_val == 0] = NA
-    levels(cur_val)[levels(cur_val) == 0] = NA
+  # Create the bubble layers
+  geom_args = list()
+  geom_args$mapping = aes(geometry=geometry, size = bubble_val)
```
Comment (capnrefsmmat) [inline R-packages/covidcast/R/plot.R:690]:
```suggestion
  geom_args$mapping = aes(geometry = geometry, size = bubble_val)
```

File: R-packages/covidcast/R/plot.R [hunk 5]
```diff
@@ -582,69 +582,130 @@ plot_bubble = function(x, time_value = NULL, include = c(), range = NULL,
...
+  geom_args$color = col
+  geom_args$alpha = alpha
+  geom_args$na.rm = TRUE
+
+  # If there is no bubble data (all values are missing), return blank layer
+  bubble_blank_if_all_na = function(geom_args, df){
```
Comment (capnrefsmmat) [inline R-packages/covidcast/R/plot.R:696]:
```suggestion
  bubble_blank_if_all_na = function(geom_args, df) {
```

File: R-packages/covidcast/R/plot.R [hunk 6]
```diff
@@ -582,69 +582,130 @@ plot_bubble = function(x, time_value = NULL, include = c(), range = NULL,
...
+    geom_args$data = df
+    bubble_layer = ggplot2::geom_blank()
+    if(!all(is.na(df$bubble_val))){
```
Comment (capnrefsmmat) [inline R-packages/covidcast/R/plot.R:699]:
```suggestion
    if (!all(is.na(df$bubble_val))) {
```

File: R-packages/covidcast/R/plot.R [hunk 7]
```diff
@@ -582,69 +582,130 @@ plot_bubble = function(x, time_value = NULL, include = c(), range = NULL,
...
+    map_df = sf::st_read(system.file(
+      "shapefiles/county/cb_2019_us_county_5m.shp",
+      package = "covidcast"),
+      quiet = TRUE)
+    map_df$STATEFP <- as.character(map_df$STATEFP)
+    map_df$GEOID <- as.character(map_df$GEOID)
+    # Get rid of megacounties
+    # Set color for observed states as white, missing as missing_col
+    # Set bubble size for all observed states
+    map_df = map_df %>% dplyr::filter(!(COUNTYFP == "000")) %>%
```
Comment (capnrefsmmat) [inline R-packages/covidcast/R/plot.R:594]:
```suggestion
    map_df = map_df %>%
      dplyr::filter(!(COUNTYFP == "000")) %>%
```

File: R-packages/covidcast/R/plot.R [hunk 8]
```diff
@@ -582,69 +582,130 @@ plot_bubble = function(x, time_value = NULL, include = c(), range = NULL,
...
 
-  # Determine which locations are missing
-  map_mis = rep(1, length(map_geo))
-  map_mis[map_geo %in% geo] = 0
-
+  # Important: make into a factor and set the levels (for the legend)
+  # Factor bubble values before splitting up into mainland and non-main layers
+  map_df$bubble_val <- factor(map_df$bubble_val, levels = breaks)
+  
+  # Explicitly drop zeros (and from levels) unless were asked not to
```
Comment (capnrefsmmat) [inline R-packages/covidcast/R/plot.R:642]:
```suggestion
  # Explicitly drop zeros (and from levels) unless we're asked not to
```

**Review by capnrefsmmat — CHANGES_REQUESTED** [review]
Looking good. Just some style suggestions and minor improvements.

I think, though, we should file a follow-up issue to make `plot_choro` and `plot_bubble` share as much of the shapefile-loading and transformation code as possible. All the `is_alaska` checks and similar appear in several places, and there's enough duplication that `plot_choro` even loads the same state shapefile twice when plotting state data. We can probably simplify quite a bit.

(but that can wait until after we fix the package by merging this PR)

File: R-packages/covidcast/R/plot.R (hunk 3 above)
Comment (sarah-colq) [reply R-packages/covidcast/R/plot.R:685]: Is this sufficient?  This knowledge is based on stack overflow answers and a visual inspection of the bubbles.

**Comment by sarah-colq** [comment]
I agree with filing a follow up issue.  I didn't want to take on too much with this PR and break everything since the dependency on usmap should be removed as soon as possible.  I'll file one.

File: R-packages/covidcast/R/plot.R (hunk 3 above)
Comment (capnrefsmmat) [reply R-packages/covidcast/R/plot.R:685]: I think the new comment is sufficient, yeah. Even if the warning is right and the centroids aren't exact, we don't really need them to be perfectly precise

**Review by capnrefsmmat — APPROVED** [review]
Looks good, thanks

**Pull request merged**